### PR TITLE
Add the FAO crop model

### PR DESCRIPTION
**Pull request by teytaud**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Motivation and Context / Related issue

Crop feeds a lot of people.

## Checklist

- [x] The documentation is up-to-date with the changes I made.
- [x] I have read the [**CONTRIBUTING**](https://facebookresearch.github.io/nevergrad/contributing.html) document and completed the CLA (see [**CLA**](https://facebookresearch.github.io/nevergrad/contributing.html#contributor-license-agreement-cla)).
- [x] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

File: nevergrad/benchmark/experiments.py [hunk 1]
```diff
@@ -21,6 +21,7 @@
 from nevergrad.functions.arcoating import ARCoating
 from nevergrad.functions import images as imagesxp
 from nevergrad.functions.powersystems import PowerSystem
+from nevergrad.functions.ac import Ac
```
Comment (jrapin) [inline nevergrad/benchmark/experiments.py:24]: can you please rename to something more intelligible? AquaCrop seems ok, NgAquaCrop to differentiate it from the base one. AC is just not clear

File: nevergrad/benchmark/experiments.py [hunk 2]
```diff
@@ -1157,6 +1158,24 @@ def realworld(seed: tp.Optional[int] = None) -> tp.Iterator[Experiment]:
...
+    funcs = [Ac(i, 300.0 + 150.0 * np.cos(i)) for i in range(3, 7)]
+    seedg = create_seed_generator(seed)
+    optims = get_optimizers("basics", seed=next(seedg))
+    for budget in [25, 50, 100, 200, 400, 800, 1600]:
+        for num_workers in [1, 30]:
+            if num_workers < budget:
+                for algo in optims:
+                    for fu in funcs:
+                        xp = Experiment(fu, algo, budget, num_workers=num_workers, seed=next(seedg))
+                        skip_ci(reason="Too slow")
```
Comment (jrapin) [inline nevergrad/benchmark/experiments.py:1174]: is it a copy paste or is it indeed too slow?

File: nevergrad/functions/ac/ac.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+from nevergrad.parametrization import parameter
+from ..base import ExperimentFunction
+
+# pylint: disable=too-many-locals,too-many-statements
+
+# Inspired by
+# https://colab.research.google.com/github/thomasdkelly/aquacrop/blob/master/tutorials/AquaCrop_OSPy_Notebook_3.ipynb#scrollTo=YDm931IGNxCb
+
+from aquacrop.classes import *
+from aquacrop.core import *
```
Comment (jrapin) [inline nevergrad/functions/ac/ac.py:21]: please avoid star imports, it makes it less clear what comes from what.

File: requirements/bench.txt [hunk 4]
```diff
@@ -33,3 +33,4 @@ olymp==0.0.1b0 ; sys_platform == "linux"
 silence_tensorflow  # for olymp
 tensorflow_probability  # for olymp
 bayes-optim==0.2.5.5
+aquacrop
```
Comment (jrapin) [inline requirements/bench.txt:36]: add a version

File: requirements/bench.txt [hunk 5]
```diff
@@ -33,3 +33,4 @@ olymp==0.0.1b0 ; sys_platform == "linux"
 silence_tensorflow  # for olymp
 tensorflow_probability  # for olymp
 bayes-optim==0.2.5.5
+aquacrop=0.2
```
Comment (jrapin) [inline requirements/bench.txt:36]: it should be either ==, or much better >=

File: nevergrad/benchmark/experiments.py (hunk 1 above)
Comment (teytaud) [reply nevergrad/benchmark/experiments.py]: Fixed.

File: requirements/bench.txt (hunk 4 above)
Comment (teytaud) [reply requirements/bench.txt]: Fixed.

File: nevergrad/functions/ac/ac.py (hunk 3 above)
Comment (teytaud) [reply nevergrad/functions/ac/ac.py]: Ok. However I am getting farther from the original code, which is a bit sad. I keep the original code in comment.

File: requirements/bench.txt (hunk 5 above)
Comment (teytaud) [reply requirements/bench.txt]: Entirely removed for the sake of compatibility.

**Review by jrapin — APPROVED** [review]

**Pull request merged**